pkg/asset/ignition/machine/node.go

```diff
@@ -36,4 +38,19 @@ func pointerIgnitionConfig(installConfig *types.InstallConfig, rootCA []byte, ro
 			},
 		},
 	}
+	if installConfig.Platform.Name() == azure.Name {
+		setAzureHostName(config)
+	}
+	return config
+}
+
+//until hostname is set by afterburn fixed
+//and included in the azure image: https://github.com/coreos/afterburn/issues/197
+func setAzureHostName(ignitionConfig *ignition.Config) {
```

Comment:
Can we keep platform-specific code in platform-specific subpackages?  I'd rather keep the cross-platform packages limited to generic stuff and then any per-platform switching necessary to call those packages.  How about:

```go
if installConfig.Platform.Name() == azure.Name {
   azure.ModifyPointerIgnitionConfig(config)
}
```

so the core API is "Azure may need to do something special, but we don't have to care what it is"?

Reply:
agreed. however, this should be a temporary workaround until afterburn has fixed the issue.

Reply:
updated as per your comment @wking 
We actually have 1 more change to fit in the ignition until the image include it (`resolv.conf`), so that's a welcome improvement